crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs

```diff
@@ -113,31 +116,84 @@ pub enum DeutschebankSEPAApproval {
 }
 
 #[derive(Debug, Serialize, PartialEq)]
-pub struct DeutschebankPaymentsRequest {
+pub struct DeutschebankMandatePostRequest {
     approval_by: DeutschebankSEPAApproval,
     email_address: Email,
     iban: Secret<String>,
     first_name: Secret<String>,
     last_name: Secret<String>,
 }
 
+#[derive(Debug, Serialize)]
+#[serde(untagged)]
+pub enum DeutschebankPaymentsRequest {
+    MandatePost(DeutschebankMandatePostRequest),
+    DirectDebit(DeutschebankDirectDebitRequest),
+}
+
 impl TryFrom<&DeutschebankRouterData<&PaymentsAuthorizeRouterData>>
     for DeutschebankPaymentsRequest
 {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
         item: &DeutschebankRouterData<&PaymentsAuthorizeRouterData>,
     ) -> Result<Self, Self::Error> {
-        let billing_address = item.router_data.get_billing_address()?;
-        match item.router_data.request.payment_method_data.clone() {
-            PaymentMethodData::BankDebit(BankDebitData::SepaBankDebit { iban, .. }) => Ok(Self {
-                approval_by: DeutschebankSEPAApproval::Click,
-                email_address: item.router_data.request.get_email()?,
-                iban,
-                first_name: billing_address.get_first_name()?.clone(),
-                last_name: billing_address.get_last_name()?.clone(),
-            }),
-            _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
+        match item
+            .router_data
+            .request
+            .mandate_id
+            .clone()
+            .and_then(|mandate_id| mandate_id.mandate_reference_id)
+        {
```

Comment:
Check if `is_mandate_payment` then only proceed with processing the sepa as mandate will be created

Reply:
Done